### PR TITLE
Pin XGBoost version

### PR DESCRIPTION
**Pull request by bchen1116**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bchen1116** [comment]
@conda-forge-admin, please rerender

**Comment by freddyaboulton** [comment]
For context @bchen1116 and I noticed that there was a missing pin for xgboost that we theorize is making the conda check ci fail on one of his prs. We're testing that theory now!

**Comment by chukarsten** [comment]
Please let me know if I messed something up...

**Comment by freddyaboulton** [comment]
@chukarsten No worries, it wasn't you!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ source:
   sha256: 2b2dc2d81b3504696fc80ee4824bec91aee49c53695467e75845910a92749ed2
 
 build:
-  number: 0
+  number: 2
```
Comment (freddyaboulton) [inline recipe/meta.yaml:12]: This should be 1?

**Comment by freddyaboulton** [comment]
@bchen1116 I think this is good to merge after we rest the `build_number` to 1!

**Pull request merged**